api/grpcserver/mesh_service.go

```diff
@@ -0,0 +1,360 @@
+			return nil, status.Errorf(codes.Internal, "error retrieving tx data")
+		}
+		for _, t := range txs {
+			res.Data = append(res.Data, &pb.AccountMeshData{
+				DataItem: &pb.AccountMeshData_Transaction{
+					Transaction: convertTransaction(t),
+				},
+			})
+		}
+	}
+
+	// Gather activation data
+	if in.Filter.AccountMeshDataFlags&uint32(pb.AccountMeshDataFlag_ACCOUNT_MESH_DATA_FLAG_ACTIVATIONS) != 0 {
+		// TODO: we have no way to look up activations by coinbase
+		// so we have no choice but to read all of them (!!!)
+		var activations []types.ATXID
+		for l := startLayer; l <= s.Mesh.LatestLayer(); l++ {
+			layer, err := s.Mesh.GetLayer(l)
+			if layer == nil || err != nil {
+				return nil, status.Errorf(codes.Internal, "error retrieving layer data")
+			}
+
+			for _, b := range layer.Blocks() {
+				activations = append(activations, b.ATXIDs...)
+			}
+		}
+
+		// Look up full data
+		atxs, matxs := s.Mesh.GetATXs(activations)
+		if len(matxs) != 0 {
+			log.Error("could not find activations %v", matxs)
+			return nil, status.Errorf(codes.Internal, "error retrieving activations data")
+		}
+		for _, atx := range atxs {
+			// Filter here, now that we have full data
+			if atx.Coinbase == addr {
+				pbatx, err := convertActivation(atx)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "error serializing activation data")
+				}
+				res.Data = append(res.Data, &pb.AccountMeshData{
+					DataItem: &pb.AccountMeshData_Activation{
+						Activation: pbatx,
+					},
+				})
+			}
+		}
+	}
+
+	res.TotalResults = uint32(len(res.Data))
+	return res, nil
+}
+
+func (s MeshService) getTxIdsFromMesh(minLayer types.LayerID, addr types.Address) []types.TransactionID {
+	var txIDs []types.TransactionID
+	for layerID := minLayer; layerID < s.Mesh.LatestLayer(); layerID++ {
+		destTxIDs := s.Mesh.GetTransactionsByDestination(layerID, addr)
+		txIDs = append(txIDs, destTxIDs...)
+		originTxIds := s.Mesh.GetTransactionsByOrigin(layerID, addr)
+		txIDs = append(txIDs, originTxIds...)
+	}
+	return txIDs
+}
+
+func convertTransaction(t *types.Transaction) *pb.Transaction {
+	return &pb.Transaction{
+		Id: &pb.TransactionId{Id: t.ID().Bytes()},
+		Data: &pb.Transaction_CoinTransfer{
+			CoinTransfer: &pb.CoinTransferTransaction{
+				Receiver: &pb.AccountId{Address: t.Recipient.Bytes()},
+			},
+		},
+		Sender: &pb.AccountId{Address: t.Origin().Bytes()},
+		GasOffered: &pb.GasOffered{
+			// TODO: Check this math! Does fee == price?
+			GasPrice:    t.Fee,
+			GasProvided: t.GasLimit,
+		},
+		Amount:  &pb.Amount{Value: t.Amount},
+		Counter: t.AccountNonce,
+		Signature: &pb.Signature{
+			// TODO: confirm default signature scheme
+			Scheme:    pb.Signature_SCHEME_ED25519_PLUS_PLUS,
+			Signature: t.Signature[:],
+			PublicKey: t.Origin().Bytes(),
+		},
+	}
+}
+
+func convertActivation(a *types.ActivationTx) (*pb.Activation, error) {
+	// TODO: It's suboptimal to have to serialize every atx to get its
```

Comment:
@antonlerner is there anything we can do about this? Do you think we should store commitment size as a property of an ATX? Or is there no choice but to calculate it each time by serializing, like this?

Reply:
I think this can be calcualted from the config, @noamnelke ?

Reply:
Opened #2095 to follow up on this